packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -327,12 +344,17 @@ class Scrollable extends StatefulWidget {
   /// If no [Scrollable] ancestor is found, then this method will assert in
   /// debug mode, and throw an exception in release mode.
   ///
+  /// The optional [Axis] allows targeting of a specific [Scrollable] of that
+  /// axis, useful when Scrollables are nested or using
+  /// [TwoDimensionalScrollable]. When [axis] is provided, the nearest enclosing
+  /// [ScrollableState] in that [Axis] is returned.
+  ///
   /// See also:
   ///
   /// * [Scrollable.maybeOf], which is similar to this method, but returns null
   ///   if no [Scrollable] ancestor is found.
-  static ScrollableState of(BuildContext context) {
-    final ScrollableState? scrollableState = maybeOf(context);
+  static ScrollableState of(BuildContext context, { Axis? axis }) {
+    final ScrollableState? scrollableState = maybeOf(context, axis: axis);
```

Comment:
Do we need to adjust the error message below to indicate that maybe there was a scrollable in the context, it just didn't have the right axis?